### PR TITLE
Implementing constructApplications API

### PR DESCRIPTION
**Pull request by joeldenning**
This api returns single-spa applications that can then be registered.

### CODE REVIEW

File: src/constructApplications.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+   * name: string;
+   * customProps: object;
+   * activeWhen: import('single-spa').Activity
+   * }>}
+   */
+  const partialApplications = Object.keys(applicationMap).map((name) => {
+    const resolvedRoutes = applicationMap[name];
+    return {
+      name,
+      customProps: (_name, location) => {
```
Comment (joeldenning) [inline src/constructApplications.js:33]: This relies on https://github.com/single-spa/single-spa/pull/533 which was released in single-spa@5.4.0

File: src/constructApplications.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        const route = resolvedRoutes.find((route) =>
+          location[routes.mode === "hash" ? "hash" : "pathname"].startsWith(
```
Comment (TheMcMurder) [inline src/constructApplications.js:35]: In the future we'll be changing this to use `pathToActiveWhen` right?

Comment (joeldenning) [inline src/constructApplications.js:35]: In a future PR, this will no longer be string `startsWith`, but will call the `pathToActiveWhen` function

Comment (joeldenning) [reply src/constructApplications.js:35]: yep!

**Pull request merged**